gridscale/resource_gridscale_server.go

```diff
@@ -810,6 +828,8 @@ func resourceGridscaleServerCreate(d *schema.ResourceData, meta interface{}) err
 
 			requestBody.HardwareProfileConfig = config
 		}
+		// if HardwareProfileConfig is set, ignore HardwareProfile
+		requestBody.HardwareProfile = ""
```

Comment:
should be better to send an error for the customer in case he configures `HardwareProfile` and `HardwareProfileConfig` instead of ignoring `HardwareProfile`.